Recon/src/Recon/PFTrackProducer.cxx

```diff
@@ -28,14 +29,39 @@ void PFTrackProducer::produce(framework::Event& event) {
   std::vector<ldmx::SimTrackerHit> pfTracks;
   if (truthTracking_) {
     for (const auto& spHit : ecalSpHits) {
-      if (spHit.getTrackID() != 1 || fabs(240 - spHit.getPosition()[2]) > 0.1 ||
-          spHit.getMomentum()[2] <= 0)
-        continue;
       if (spHit.getPdgID() == 22 || spHit.getPdgID() == 2112) continue;
-      pfTracks.push_back(spHit);
-      break;
-    }
-  }
+      if (fabs(240 - spHit.getPosition()[2]) > 0.1) continue;
+      if (doElectronTracking_) {  // only select electron SP hits
+        if (spHit.getPdgID() != 11) continue;
+        if (spHit.getTrackID() < 2 && spHit.getMomentum()[2] > 2500) {
```

Comment:
Why is this 2500? This should be configurable, and then have it the incoming energy dependent, right?